web/testing/regression-tests/src/test-runner.js

```diff
@@ -0,0 +1,289 @@
+// test-runner
+
+const KEYBOARDS_RELATIVE_PATH = "/keyboards/"; 
+const TEST_BATCH_SIZE = 100; // or Infinity
+
+var receiver;
+
+var windowLoad = new Promise(function(resolve, reject) {
+  window.addEventListener('load', function() {
+    // Initialise KeymanWeb
+    keyman.init({
+      attachType: 'auto',
+      resources: '/web/'
```

Comment:
It may be worth pointing out that `keyman.init` itself returns a `Promise` now; it may be best to put the follow-up code within that `Promise`'s `then` clause to ensure everything happens in the right order.

Unless I'm mistaken, it's quite possible for `attachToControl` and `setKeyboardForControl` to be run before KMW properly initializes in the code's current state.

Reply:
Which version of KeymanWeb started using the promise? I need to test against v10 stable at least.

Reply:
OK, fair.  That'd be v11.  You can probably get by if you include the code after `init`, since it's in a handler that's listening for a body.load event.